controllers/etcd_controller.go

```diff
@@ -159,30 +179,159 @@ func (r *EtcdReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		}
 	}
 	if err := r.addFinalizersToDependantSecrets(etcdCopy); err != nil {
-		if err := r.updateEtcdErrorStatus(etcdCopy, err); err != nil {
+		if err := r.updateEtcdErrorStatus(etcd, etcdCopy, err); err != nil {
 			return ctrl.Result{
 				Requeue:      true,
 				RequeueAfter: time.Second * 5,
-			}, nil
+			}, err
 		}
 	}
 
-	imageVector, err := imagevector.ReadGlobalImageVectorWithEnvOverride(imageYAMLPath)
-	if err != nil {
+	if err := r.reconcileEtcd(etcdCopy); err != nil {
+		if err := r.updateEtcdErrorStatus(etcd, etcdCopy, err); err != nil {
+			return ctrl.Result{
+				Requeue:      true,
+				RequeueAfter: time.Second * 5,
+			}, err
+		}
 		return ctrl.Result{
 			Requeue:      true,
 			RequeueAfter: time.Second * 5,
 		}, err
 	}
 
-	images, err := imagevector.FindImages(imageVector, []string{"etcd", "etcd-backup-restore"})
-	if err != nil {
+	if err := r.updateEtcdStatus(etcd); err != nil {
 		return ctrl.Result{
 			Requeue:      true,
 			RequeueAfter: time.Second * 5,
 		}, err
 	}
 
+	logger.Infof("Succesfully reconciled etcd: %s", etcd.GetName())
+
+	return ctrl.Result{
+		Requeue:      true,
+		RequeueAfter: time.Second * 5,
```

Comment:
Should the requeue rate be the same for success as well as error?

Reply:
Changed it to 5 minutes.